packages/rust-bindings/src/aux_functions.rs

```diff
@@ -24,6 +33,88 @@ pub struct CredId {
     pub cred_id: ExampleCurve,
 }
 
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IdRequestInput {
+    ip_info:        IpInfo<Bls12>,
+    global_context: GlobalContext<ExampleCurve>,
+    ars_infos:      BTreeMap<ArIdentity, ArInfo<ExampleCurve>>,
+    seed:           String,
+    net:            String,
+    identity_index: u32,
+    ar_threshold:   Option<u8>,
+}
+
+pub fn create_id_request_v1_aux(input: IdRequestInput) -> Result<String> {
+    let seed_decoded = hex::decode(&input.seed)?;
+    let seed: [u8; 64] = match seed_decoded.try_into() {
+        Ok(s) => s,
+        Err(_) => bail!("The provided seed {} was not 64 bytes", input.seed),
+    };
+
+    let net = match input.net.as_str() {
+        "Mainnet" => Net::Mainnet,
+        "Testnet" => Net::Testnet,
+        _ => bail!("Unknown net"),
+    };
+    let wallet = ConcordiumHdWallet { seed, net };
+
+    let prf_key: prf::SecretKey<ArCurve> = wallet.get_prf_key(input.identity_index)?;
+
+    let id_cred_sec: PedersenValue<ArCurve> =
+        PedersenValue::new(wallet.get_id_cred_sec(input.identity_index)?);
+    let id_cred: IdCredentials<ArCurve> = IdCredentials { id_cred_sec };
+
+    let sig_retrievel_randomness: ps_sig::SigRetrievalRandomness<Bls12> =
+        wallet.get_blinding_randomness(input.identity_index)?;
+
+    let num_of_ars = input.ars_infos.len();
+    let threshold = match input.ar_threshold {
+        Some(threshold) => {
+            ensure!(threshold > 0, "arThreshold must be at least 1.");
+            ensure!(
+                num_of_ars >= usize::from(threshold),
+                "Number of anonymity revokers in arsInfos should be at least arThreshold."
+            );
+            Threshold(threshold)
+        }
+        None => {
+            // arThreshold not specified, use `number of anonymity revokers` - 1 or 1 in the
+            // case of only a single anonymity revoker.
+            ensure!(
+                num_of_ars > 0,
+                "arsInfos should have at least 1 anonymity revoker."
+            );
+            Threshold(max((num_of_ars - 1).try_into().unwrap_or(255), 1))
+        }
+    };
+
+    let chi = CredentialHolderInfo::<ArCurve> { id_cred };
+
+    let aci = AccCredentialInfo {
+        cred_holder_info: chi,
+        prf_key,
+    };
+
+    // Choice of anonymity revokers, all of them in this implementation.
```

Comment:
This comment does not seem to fit in here.

Reply:
Yeah, I removed it. It's just the notion that we are putting all ars' infos into the context. But if we were to change this, it would be outside the rust code anyway.